File: dot.net/Ogman.Internals/Ogman.Internals.Tests/PublicFacingTests.cs
using System.Text;
using NUnit.Framework;

namespace Ogman.Internals.Tests
{
    [TestFixture]
    public class PublicFacingTests
    {
        public class Derived : PublicFacing
        {
            public Derived(StringBuilder output)
                : base(output)
            {
            }

            /*this is not allowed at compile time due non existing virtual method called DoInternalWork*/
            //public override void DoInternalWork()
            //{
            //    
            //}
        }

        [Test]
        public void Test()
        {
            var output = new StringBuilder();
            var inst = new PublicFacing(output);
            inst.DoWork();
            /*this is not allowed at compile time due to low visibility*/
            //inst.DoInternalWork();
            Assert.That(output.ToString(),
                        Is.EqualTo("Public work is based on internal work."));
        }
    }
}

File: java/org.ogman.internals/tests/org/ogman/internals/tests/PublicFacingTests.java
package org.ogman.internals.tests;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import org.junit.Test;
import org.ogman.internals.PublicFacing;

public class PublicFacingTests {

	public class Derived extends PublicFacing {
		public Derived(StringBuffer output) {
			super(output);
		}

		/*this is not allowed at compile time due to low visibility*/
		//@Override
		//void doInternalWork(){
		//	
		//}
	}
	
	@Test
	public void test() {
		StringBuffer output = new StringBuffer();
		PublicFacing inst = new PublicFacing(output);
		inst.doWork();
		/*this is not allowed at compile time due to low visibility*/
		//inst.doInternalWork();
		assertThat(output.toString(),
				is("Public work is based on internal work."));
	}
}
